lib/io.js

```diff
@@ -357,9 +353,10 @@ DatumReader.prototype = {
         var oldOffset = decoder._input.offset;
         var schemaIndex = decoder.readLong();
 
-        if (schemaIndex < 0 || schemaIndex >= writersSchema.schemas.length) {
```

Comment:
What was wrong about this conditional? It seems that by leaving it using native operators, you can remove most calls to `toNumber()` from the entire file. Or is it the case that `readLong()` is called externally and always expects to return a `Long` object?